packages/golden_toolkit/lib/src/testing_tools.dart

```diff
@@ -216,7 +220,29 @@ Future<void> screenMatchesGolden(
   }
 }
 
-// Matches Golden file is the easiest way for the images to be requested.
-Future<void> _primeImages(String fileName, Finder finder) => matchesGoldenFile(fileName).matchAsync(finder);
+/// A function that primes all assets by calling [matchesGoldenFile] with the tester.
+/// Doing so may have some unwanted side effects like the creation of a failures folder even when the test
+/// succeeds.
+///
+/// See also:
+/// * [GoldenToolkitConfiguration.primeAssets] to configure a global asset prime function.
+Future<void> defaultPrimeAssets(WidgetTester tester, String name, Finder finder) =>
+    matchesGoldenFile(name).matchAsync(finder);
```

Comment:
so I think we can simplify the prime assets api... 

this old version was such a hack, that I don't think the filename OR the finder actually matter...

I believe we can remove them from the signature, and just hardcode the filename to a constant, and the finder to be the root of the widget tree. That might make the signature more palatable 😄

Reply:
> and just hardcode the filename to a constant

This would result in a file always being created. I will investigate why this hack works and implement sth. similar that does not need a fileName and thus does not create a file.

Reply:
Ok the legacy version does not work for huge images. Seems like calling mathcesGoldenFile only triggers the decompression and decoding of all images. For small images this may work because they finish decoding before the actual call but for bigger images this does not work reliable. @coreysprague Do you think a test that verifies that the new implementation works even for huge images would be useful? This would probably add another 5mb to the repo.

Reply:
Ok now I realized that matchesGoldenFIle just consumes some time which is enough for all the small images to be decoded. I ran a benchmark and matchesGoldenFile took an average of 25ms. Waiting for the same duration has the same effect... The time spent in the `matchesGoldenFile` call should be relative to the time needed to load small images. I suggest just calling the heavy operations done by the function to waste some time in the legacy implementation.

Reply:
Waiting for a const Duration should be very flaky because we dont know the speed of the device the test is running on.

Reply:
Just commited `legacyPrimeAssets` which just wastes time by rasterizing the next RepaintBoundary and converting the result image to a png.

Reply:
@christian-muertz you are correct. The old one triggered rasterization and essentially just wasted cycles, and generally worked, but not all the time. 

I'm not too worried about the large image test for the new solution. The new solution is not dependent on winning a non-deterministic race. I think your proposal for an alternate legacy solution makes sense to me. This legacy implementation should be short lived anyway. I mostly just want to give consumers a release or two to migrate in case there are issues with the new solution.